### DISCUSSION TITLE
wis2box 1.0b8 released

### DISCUSSION THREAD

**Discussion by tomkralidis** [wmo-im/wis2box#755] [Announcements]
This is the eighth 1.0 beta release of wis2box. This release includes a new CAP XML data pipeline, added utilities for WIS2 data download, as well numerous bug fixes and updates for WIS2 compliance.

Users are strongly encouraged to update to this release for WIS2 compliance.

## Enhancements/updates
- add CAP XML data pipeline
- add data download functionality (via wis2downloader)
- add support for GTS headers applied in WIS2 Notification message
- adds discovery metadata republish command in CLI (includes automated daily publishing job)
- add station association capability in CLI
- show all system warnings (previously set to errors only)
- dataset driven approach implemented in wis2box data automated ingest and CLI
- improved logging / logfile management
- improved disk usage monitoring and management
- update MinIO container version
- add UI vulnerability testing for wis2box-webapp and wis2box-ui
- updated help links for FM-12 and CSV to BUFR in wis2box-webapp
- add metadata_id (for all data) and security (for recommended data) elements to WIS2 Notifications
- store local copy of WIGOS station metadata lookups
- add new discovery metadata templates in wis2box-webapp (drifters and ship based surface weather observations)
- supporting documentation for configuring email alerts from Grafana dashboard
- improved management of API configuration in wis2box-api

## Bug fixes
- fix WCMP2 output themes/concepts formatting
- store local copy of WIGOS station metadata lookups for stability
- ensure wis2box starts successfully on Docker host restart
- add test to ensure `WIS2BOX_BROKER_PUBLIC` is correctly configured

## New installations

Please follow the instructions provided at https://docs.wis2box.wis.wmo.int/en/1.0b8/user/getting-started.html

## Migrating existing installations

When updating an existing wis2box instance, please run the following commands from the current release directory:

```bash
# stop wis2box
python3 wis2box-ctl.py stop

# delete old Docker images
docker image prune -a

# copy wis2box.env from the old to the new release directory
cp ~/wis2box-1.0b7/wis2box.env ~/wis2box-1.0b8/wis2box.env
```

wis2box-1.0b8 includes new/added download functionality. To use this functionality, run the following commands:

```bash
# set the DOCKER_GID environment variable
echo "DOCKER_GID=$(getent group docker | cut -d: -f3)" >> ~/wis2box-1.0b8/wis2box.env

# and add the "/downloads" directory to the directory defined as your WIS2BOX_HOST_DATADIR and set the permission to the docker group:
export $(grep -v '^#' ~/wis2box-1.0b8/wis2box.env | xargs -d '\n')
mkdir -p ${WIS2BOX_HOST_DATADIR}/downloads
chown -R $(whoami):docker ${WIS2BOX_HOST_DATADIR}/downloads
chmod -R 775 ${WIS2BOX_HOST_DATADIR}/downloads
```

Go to the new wis2box release directory and start wis2box:
```bash
cd ~/wis2box-1.0b8
python3 wis2box-ctl.py start
```

## Documentation
Documentation for this release is available at https://docs.wis2box.wis.wmo.int/en/1.0b8

## Feedback
All contributions are always welcome. Feedback is welcome via GitHub [issues](https://github.com/wmo-im/wis2box/issues) or [discussions](https://github.com/wmo-im/wis2box/discussions).

The wis2box team

https://docs.wis2box.wis.wmo.int

**Comment by pgulb** [comment]
Is wis2downloader using local file storage? Should not it use s3 as well?

**Reply by maaikelimper** [reply]
it currently writes to local disk, you can request and/or contribute to write to S3-storage via the github-repo here: https://github.com/wmo-im/wis2downloader

**Reply by pgulb** [reply]
Thank you, will consider it
